File: javascript/matrices/5-Average.js
// Write a program that reads a 3x3 matrix and calculates the average
// of the values present in the even positions (sum of the even indices) of the matrix.

const prompt = require('prompt-sync')({ sigint: true });

const matrix = [];
let sum = 0;
let count = 0;

// Fill matrix with user input
for (let i = 0; i < 3; i++) {
  matrix[i] = [];
  for (let j = 0; j < 3; j++) {
    matrix[i][j] = parseInt(prompt(`Enter a value for matrix[${i}][${j}]: `));
    if ((i + j) % 2 === 0) {
      sum += matrix[i][j];
      count++;
    }
  }
}

// Calculate average of values at even positions
const average = count !== 0 ? sum / count : 0;

console.log('Matrix:', matrix);
console.log('The average of values at even positions:', average);

File: python/matrices/5-Average.py
# Write a program that reads a 3x3 matrix and calculates the average
# of the values present in the even positions (the sum of the even
# indices) of the matrix.

# Initialize a 3x3 matrix
matrix = []

# Read the matrix from a user
print("Enter the values of the 3x3 matrix:")
for _ in range(3):
  row = []
  for _ in range(3):
    value = int(input("Enter a value: "))
    row.append(value)
  matrix.append(row)

# Calculate the sum and count of the values at even positions
sum_even = 0
count_even = 0

for i in range(3):
  for j in range(3):
    if (i + j) % 2 == 0: # Check if the sum of indices is even
      sum_even += matrix[i][j]
      count_even += 1

# Calculate the average of values at even positions
average_even = sum_even / count_even

# Display the result
print("The average of values at even positions:", average_even)